server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/HAProxyMessageHandler.java

```diff
@@ -68,8 +68,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 LOGGER.info("Connection from {} runs through {} proxy", haproxyMsg.sourceAddress(), haproxyMsg.destinationAddress());
                 // Refresh MDC info to account for proxying
                 MDCBuilder boundMDC = IMAPMDCContext.boundMDC(ctx);
+                Flux.fromIterable(connectionChecks).concatMap(connectionCheck -> connectionCheck.validate(InetSocketAddress.createUnresolved(haproxyMsg.sourceAddress(), haproxyMsg.sourcePort()))).then().block();
```

Comment:
Why not just `InetSocketAddress abcxyz = new InetSocketAddress(haproxyMsg.sourceAddress(), haproxyMsg.sourcePort())` and reuse it above also to recalculate the variable again?